packages/core/src/base.test.ts

```diff
@@ -121,6 +121,35 @@ describe("base class tests", () => {
     expect(base.synced).toEqual(1);
   });
 
+  it("should resolve promise after retrying unsuccessfully", async () => {
```

Comment:
The test passes even before the change in `base.ts`. Bothe before and after triggers a Jest warning:

```
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
  console.error packages/core/src/base.ts:255
    [ { dt: 2023-07-10T13:16:01.821Z, level: 'info', message: 'test' } ]
```

I would suggest merging https://github.com/logtail/logtail-js/pull/66 instead and reverting the commit in this PR.